src/main/java/commission/service/SaleServiceImpl.java

```diff
@@ -27,34 +31,27 @@ public List<Sale> findAll(){
     }
     @Override
     public void insertSale(Sale sale){
-        saleDao.insertSale(sale);
+        saleDao.insertSale(sale, commissionPerSale(sale));
     }
 
+
     @Override
     public Optional<Sale> findSaleById(long id) {
-        var temp = saleDao.findSaleById(id);
-        Sale sale = null;
-        if (temp.isPresent()) {
-            sale = temp.get();
-        } else {
-            throw new ResponseStatusException(NOT_FOUND, "ID not found");
+        try{
+            return saleDao.findSaleById(id);
+        } catch (EmptyResultDataAccessException e){
+            logger.error("ID " + id + " not found");
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
         }
-        return Optional.of(sale);
-
-//        try {
-//            var temp = saleDao.findSaleById(id);
-//            return Optional.of(temp.get());
-//        } catch (EmptyResultDataAccessException e) {
-//            throw new RuntimeException("Invalid ID");
-//        }
     }
 
+
     @Override
     public boolean deleteSale(long id) {
         boolean isDeleted = false;
         var temp = findSaleById(id);
         if (temp.isEmpty()){
-            System.out.println("ID does not exists");
+            logger.error("ID " + id + " not found");
```

Comment:
This can't ever execute, because the `findSaleById` method throws an exception when nothing is found in the database. See my other comment about moving the exception handling to the DAO

Reply:
Moved the main error handling in DAO and I left some checks in service layer. Again not sure if its the right way.